File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    
    q = deque(progresses)
    speeds = deque(speeds)
    
    while True:
        cnt = 0
        
        if len(q) == 0: break
        
        for i, s in enumerate(speeds):
            q[i] += s
                
        if q[0] >= 100:
            while True:
                if len(q) == 0 or q[0] < 100: break
                
                q.popleft()
                speeds.popleft()
                cnt += 1
            answer.append(cnt)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {

        List<Integer> answer = new ArrayList<>();
        
        Queue<Integer> daysQ = new LinkedList<>();
        
        for(int i = 0; i < progresses.length; i++) {
            if ((100 - progresses[i])%speeds[i] == 0)
                daysQ.offer((100 - progresses[i])/speeds[i]);
            else
                daysQ.offer((100 - progresses[i])/speeds[i] + 1);
        }
        
        int tmp = daysQ.poll();
        int cnt = 1;
            
        while(!daysQ.isEmpty()) {
            
            int tmp2 = daysQ.poll();
            if(tmp >= tmp2) cnt++;
            else {
                tmp = tmp2;
                answer.add(cnt);
                cnt = 1;
            }
        }
        answer.add(cnt);
        
        return answer;
    }
}